Update intent gossip spec with proposition of implementation (#334)
anoma/anoma: tech-specs/src/explore/design/intent_gossip/topic.md

# Topic

> A topic is string and a encoding that describes this sub-network. In a topic all

Reviewer: ```suggestion
A topic is string and an encoding that describes this sub-network. In a topic all
```

intents use the exact same encoding. That encoding is known by matchmakers so it

> can match them. Whenever a node subscribes to a new topic it informs all

Reviewer: ```suggestion
can decode them to find matches. Whenever a node subscribes to a new topic it informs all
```

connected nodes and each of them propagate it. With this it’s easy to create new
topics in the intent gossip network and inform others.

> A node opt-in to a propagated new topic with the help of a filter. This filter

Reviewer: ```suggestion
Other nodes can choose to subscribe to a new topic with the help of a filter. This filter
```

> is define as a combination of a whitelist, a regex expression, and a maximum

Reviewer: ```suggestion
is defined as a combination of a whitelist, a regex expression, and a maximum
```

# Topic

A topic is string and a encoding that describes this sub-network. In a topic all
intents must use the exact same encoding. That encoding is known by matchmakers
so it can match them.  Whenever a node subscribes to a new topic it informs all

> connected nodes. Each of them tries to subscribe to it and continue the

Reviewer 2: Other nodes automatically also subscribe? Shouldn't they opt-in?

Author: A node can express an topic filter to prevent subscription to undesired topic. This filter is composed of a whitelist, a regex and a maximum subscription. 

This I describe in the next paragraph, I'm going to reformulate this sentence to express that
